services/src/main/java/org/exoplatform/officeonline/OfficeOnlineEditorService.java

```diff
@@ -92,11 +175,49 @@ public void stop() {
   }
 
   /**
-   * Adds the plugin.
+   * Node by UUID.
+   *
+   * @param workspace the workspace
+   * @param uuid the UUID
+   * @return the node
+   * @throws RepositoryException the repository exception
+   */
+  protected Node nodeByUUID(String workspace, String uuid) throws RepositoryException {
+    SessionProvider sp = sessionProviders.getSessionProvider(null);
+    Session userSession = sp.getSession(workspace, jcrService.getCurrentRepository());
+    return userSession.getNodeByUUID(uuid);
+  }
+
+  /**
+   * Can edit document.
+   *
+   * @param node the node
+   * @return true, if successful
+   * @throws RepositoryException the repository exception
+   */
+  public boolean canEditDocument(Node node) throws RepositoryException {
+    boolean res = false;
+    if (node != null) {
+      String remoteUser = WCMCoreUtils.getRemoteUser();
```

Comment:
Ensure this will return an username in all cases: WebUI request, REST/Servlet request.